app/api/db/models/user_models.py

```diff
@@ -28,7 +28,9 @@ class User(Base, IdMixin, TimestampMixin):
     date_of_birth: date = Column(Date, nullable=False)
     is_active: bool = Column(Boolean, nullable=False)
 
-    roles: Optional[list["Role"]] = relationship("Role", back_populates="user")
+    roles: Optional[list["Role"]] = relationship(
+        "Role", back_populates="user", order_by="Role.type"
```

Comment:
the `order_by` makes the order of the roles deterministic, something I caught when adding the various combinations of patching roles. previously if a user started out with `USER` role and added `ADMIN` the order will be `USER`, `ADMIN`, this ensures the order of the roles is always the same